nav2_behavior_tree/plugins/action/remove_in_collision_goals_action.cpp

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) 2024 Angsa Robotics
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <string>
+#include <memory>
+#include <limits>
+
+#include "nav2_behavior_tree/plugins/action/remove_in_collision_goals_action.hpp"
+#include "nav2_behavior_tree/bt_utils.hpp"
+#include "tf2/utils.h"
+#include "tf2_geometry_msgs/tf2_geometry_msgs.hpp"
+
+namespace nav2_behavior_tree
+{
+
+RemoveInCollisionGoals::RemoveInCollisionGoals(
+  const std::string & service_node_name,
+  const BT::NodeConfiguration & conf)
+: BtServiceNode<nav2_msgs::srv::GetCosts>(service_node_name, conf,
+    "/global_costmap/get_cost_global_costmap")
```

Comment:
why hard coded now?

Reply:
As far as I understand that's the way to do it with BtServiceNode. This is not hardcoded, it can still be overridden, see [here](https://github.com/ros-navigation/navigation2/blob/746bb880a775201fbdc4cca38a873cc3d0d9ab40/nav2_behavior_tree/include/nav2_behavior_tree/bt_service_node.hpp#L72). I must admit I didn't test overriding it

Reply:
Oh, yes, that works!